go/process/health_dao.go

```diff
@@ -147,10 +167,18 @@ func expireAvailableNodes() error {
 			`,
 		registrationPollSeconds*2,
 	)
-	return log.Errore(err)
+	if err != nil {
+		log.Errorf("ExpireAvailableNodes: failed to remove old entries: %+v", err)
+	}
+
+	// show we're done
+	expiryMutex.Lock()
+	expiryRunning = false
+	expiryMutex.Unlock()
```

Comment:
To be honest I've seen pile ups and things in the past and am just paranoid. Overly paranoid? maybe but I think that most of this type of event should probably only run once and if it doesn't then we should see there's an issue. I don't have time to check/fix existing code but if I bump into new code then I'd prefer to catch this.  If you're concerned this is too much then I guess this code can be removed but I don't see any harm in keeping it.